src/org/zaproxy/zap/extension/pscan/PassiveScanThread.java

```diff
@@ -141,7 +146,13 @@ public void run() {
 								if (shutDown) {
 									return;
 								}
-								if (scanner.isEnabled() && scanner.appliesToHistoryType(href.getHistoryType())) {
+								int hrefHistoryType = href.getHistoryType();
+								if (pscanOptions.isScanFuzzerMessages()) {
```

Comment:
It should also remove them (when disabled), but instead of doing that here it can be called by `PassiveScanParam` when changed.